### DISCUSSION TITLE
Doubtt inWifi_station program

### DISCUSSION THREAD

**Discussion by Mowitra** [Mair/esp32-course#131] [Course Qeustions] [answered]
Sir when I compile this program(https://vimeo.com/user43888278?embedded=true&source=video_title&owner=43888278) it shows the following error...how to clear this error

assert failed: xEventGroupWaitBits event_groups.c:371 (uxBitsToWaitFor != 0)

Backtrace:0x400819f2:0x3ffc4c100x400883b9:0x3ffc4c30 0x4008ef25:0x3ffc4c50 0x4008b232:0x3ffc4d70 0x400d5e89:0x3ffc4da0 0x400d5c0c:0x3ffc4e50 0x4008b545:0x3ffc4e70 
0x400819f2: panic_abort at C:/ESP_IDF/esp-idf/components/esp_system/panic.c:402

0x400883b9: esp_system_abort at C:/ESP_IDF/esp-idf/components/esp_system/esp_system.c:128

0x4008ef25: __assert_func at C:/ESP_IDF/esp-idf/components/newlib/assert.c:85

0x4008b232: xEventGroupWaitBits at C:/ESP_IDF/esp-idf/components/freertos/event_groups.c:371 (discriminator 1)

0x400d5e89: wifi_connect_sta at C:\ESP_IDF_PROJECTS\wifi_connect_1\build/../components/wifi/connect.c:68

0x400d5c0c: wifi_connect at C:\ESP_IDF_PROJECTS\wifi_connect_1\build/../main/main.c:14

0x4008b545: vPortTaskWrapper at C:/ESP_IDF/esp-idf/components/freertos/port/xtensa/port.c:131

ELF file SHA256: ac811bf4e2500406

Rebooting...
ets Jun  8 2016 00:22:57

rst:0xc (SW_CPU_RESET),boot:0x13 (SPI_FAST_FLASH_BOOT)
configsip: 0, SPIWP:0xee
clk_drv:0x00,q_drv:0x00,d_drv:0x00,cs0_drv:0x00,hd_drv:0x00,wp_drv:0x00
mode:DIO, clock div:2
load:0x3fff0030,len:6608
load:0x40078000,len:14788
ho 0 tail 12 room 4
load:0x40080400,len:3792
0x40080400: _init at ??:?

entry 0x40080694
I (29) boot: ESP-IDF v4.4.1 2nd stage bootloader
I (29) boot: compile time 11:34:44
I (29) boot: chip revision: 1
I (32) boot_comm: chip revision: 1, min. bootloader chip revision: 0
I (39) boot.esp32: SPI Speed      : 40MHz
I (43) boot.esp32: SPI Mode       : DIO
I (48) boot.esp32: SPI Flash Size : 2MB
I (52) boot: Enabling RNG early entropy source...
I (58) boot: Partition Table:
I (61) boot: ## Label            Usage          Type ST Offset   Length
I (69) boot:  0 nvs              WiFi data        01 02 00009000 00006000
I (76) boot:  1 phy_init         RF data          01 01 0000f000 00001000
I (84) boot:  2 factory          factory app      00 00 00010000 00100000
I (91) boot: End of partition table
I (95) boot_comm: chip revision: 1, min. application chip revision: 0
I (102) esp_image: segment 0: paddr=00010020 vaddr=3f400020 size=13340h ( 78656) map
I (140) esp_image: segment 1: paddr=00023368 vaddr=3ffb0000 size=0380ch ( 14348) load
I (145) esp_image: segment 2: paddr=00026b7c vaddr=40080000 size=0949ch ( 38044) load
I (161) esp_image: segment 3: paddr=00030020 vaddr=400d0020 size=6d15ch (446812) map
I (323) esp_image: segment 4: paddr=0009d184 vaddr=4008949c size=0ae88h ( 44680) load
I (342) esp_image: segment 5: paddr=000a8014 vaddr=50000000 size=00010h (    16) load
I (352) boot: Loaded app from partition at offset 0x10000
I (352) boot: Disabling RNG early entropy source...
I (364) cpu_start: Pro cpu up.
I (364) cpu_start: Starting app cpu, entry point is 0x40081184
0x40081184: call_start_cpu1 at C:/ESP_IDF/esp-idf/components/esp_system/port/cpu_start.c:160

I (350) cpu_start: App cpu up.
I (379) cpu_start: Pro cpu start user code
I (379) cpu_start: cpu freq: 160000000
I (379) cpu_start: Application information:
I (383) cpu_start: Project name:     wifi_connect_1
I (389) cpu_start: App version:      1
I (393) cpu_start: Compile time:     Jul 14 2022 11:34:02
I (399) cpu_start: ELF file SHA256:  ac811bf4e2500406...
I (405) cpu_start: ESP-IDF:          v4.4.1
I (410) heap_init: Initializing. RAM available for dynamic allocation:
I (417) heap_init: At 3FFAE6E0 len 00001920 (6 KiB): DRAM
I (423) heap_init: At 3FFB74A8 len 00028B58 (162 KiB): DRAM
I (430) heap_init: At 3FFE0440 len 00003AE0 (14 KiB): D/IRAM
I (436) heap_init: At 3FFE4350 len 0001BCB0 (111 KiB): D/IRAM
I (442) heap_init: At 40094324 len 0000BCDC (47 KiB): IRAM
I (450) spi_flash: detected chip: generic
I (453) spi_flash: flash io: dio
W (457) spi_flash: Detected size(4096k) larger than the size in the binary image header(2048k). Using the size in the binary image header.
I (471) cpu_start: Starting scheduler on PRO CPU.
I (0) cpu_start: Starting scheduler on APP CPU.
I (569) wifi:wifi driver task: 3ffbfdb8, prio:23, stack:6656, core=0
I (569) system_api: Base MAC address is not set
I (569) system_api: read default base MAC address from EFUSE
I (589) wifi:wifi firmware version: 63017e0
I (589) wifi:wifi certification version: v7.0
I (589) wifi:config NVS flash: enabled
I (589) wifi:config nano formating: disabled
I (589) wifi:Init data frame dynamic rx buffer num: 32
I (599) wifi:Init management frame dynamic rx buffer num: 32
I (599) wifi:Init management short buffer num: 32
I (609) wifi:Init dynamic tx buffer num: 32
I (609) wifi:Init static rx buffer size: 1600
I (619) wifi:Init static rx buffer num: 10
I (619) wifi:Init dynamic rx buffer num: 32
I (629) wifi_init: rx ba win: 6
I (629) wifi_init: tcpip mbox: 32
I (629) wifi_init: udp mbox: 6
I (639) wifi_init: tcp mbox: 6
I (639) wifi_init: tcp tx win: 5744
I (639) wifi_init: tcp rx win: 5744
I (649) wifi_init: tcp mss: 1440
I (649) wifi_init: WiFi IRAM OP enabled
I (659) wifi_init: WiFi RX IRAM OP enabled
I (659) phy_init: phy_version 4670,719f9f6,Feb 18 2021,17:07:07
I (769) wifi:mode : sta (84:cc:a8:79:d4:64)
I (769) wifi:enable tsf

assert failed: xEventGroupWaitBits event_groups.c:371 (uxBitsToWaitFor != 0)

**Comment by Mair** [comment, marked as answer]
Hi @Mowitra from the stack trace it appears there is an error on your code at line 68 at  /components/wifi/connect.c
its hard to tell what you are differently to what I am doing.
If you look under the video you will see the link to the github repo.

**Reply by Mowitra** [reply]
ok sir I got the output
